packages/auth-providers/clerk/api/src/decoder.ts

```diff
@@ -34,3 +37,39 @@ export const authDecoder: Decoder = async (token: string, type: string) => {
     return Promise.reject(error)
   }
 }
+
+export const jwtAuthDecoder: Decoder = async (token: string, type: string) => {
+  if (type !== 'clerk') {
+    return null
+  }
+
+  const { verifyToken } = await import('@clerk/clerk-sdk-node')
+
+  try {
+    const issuer = (iss: string) =>
+      iss.startsWith('https://clerk.') || iss.includes('.clerk.accounts')
+
+    const jwtPayload = await verifyToken(token, {
+      issuer,
+      apiUrl: process.env.CLERK_API_URL || 'https://api.clerk.dev',
+      jwtKey: process.env.CLERK_JWT_KEY,
+      apiKey: process.env.CLERK_API_KEY,
+      secretKey: process.env.CLERK_SECRET_KEY,
+    })
+
+    if (!jwtPayload.sub) {
+      return Promise.reject(new Error('Session invalid'))
+    }
+
+    // @ts-expect-error need to type this
+    const roles = jwtPayload.sessionClaims.publicMetadata['roles'] ?? []
+
+    return {
+      ...jwtPayload,
```

Comment:
This is pretty similar to what I have. The only difference from my implementation is that I only return this:

``` 
   return {
      id: jwtPayload.sub,
    }
```
    
So this might break code that depends on `authDecoder` returning an object with `{id: clerkUserId}`.

Reply:
Great call @asher-eastham, added this property back to the return in https://github.com/redwoodjs/redwood/pull/8642/commits/22a8d4630836d0230349b419ac5b375e8937ee78

Reply:
Quick follow up note for myself: figure out if this property, if customized, is the same as the sub:

<img width="558" alt="image" src="https://github.com/redwoodjs/redwood/assets/32992335/438ca8f9-da97-4058-9be6-c7534ac1b3dc">